src/main/java/org/commcare/modern/engine/cases/RecordObjectCache.java

```diff
@@ -0,0 +1,32 @@
+package org.commcare.modern.engine.cases;
+
+import org.commcare.cases.model.Case;
+import org.commcare.cases.query.QueryCache;
+
+import java.util.HashMap;
+
+/**
+ * A straightforward cache object query cache. Stores objects by their record ID.
+ *
+ * Used by other optimizations to isolate doing bulk loads and ensure that they are relevant
+ * when they occur
+ *
+ * Created by ctsims on 6/22/2017.
+ */
+
+public class RecordObjectCache<T> implements QueryCache {
```

Comment:
I'm a bit confused by this class's purpose. I don't see it ever being instantiated and I don't see where the `cachedRecordObjects` are ever written. Could you clarify?